src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java

```diff
@@ -47,9 +44,8 @@
 
     protected final String builtinName;
 
-    @CompilationFinal private FrameSlot thisSlot;
-
-    @CompilationFinal private boolean thisSlotInitialized = false;
+    private static final NodeLibrary NODE_LIBRARY = NodeLibrary.getUncached();
+    private static final InteropLibrary INTEROP = InteropLibrary.getUncached();
```

Comment:
using uncached interop libraries is quite bad for performance: couldn't you store them in a node and use the cached version instead? (see e.g. https://github.com/oracle/graal/blob/master/truffle/docs/InteropMigration.md)

Reply:
I added the caching in 531c7e9 and also brought back the old optimization that reads `<this>` from the frame slots (which does trigger in our tests). This seemed to shave 2-3 seconds off the `lodash` test when I tested it. Unfortunately that's not a proper benchmark, maybe running the V8 tests with the compiler would be a better option.

Reply:
@eleinadani does that caching look okay?

Reply:
uh, I see... let's keep it like that then. I think I should do a full review of the entire codebase and look into how interop is used anyway at some point